server/src/main/java/io/druid/metadata/SQLMetadataConnector.java

```diff
@@ -197,9 +197,10 @@ public void createPendingSegmentsTable(final String tableName)
                 + "  \"end\" VARCHAR(255) NOT NULL,\n"
                 + "  sequence_name VARCHAR(255) NOT NULL,\n"
                 + "  sequence_prev_id VARCHAR(255) NOT NULL,\n"
+                + "  sequence_name_prev_id_sha1 VARCHAR(255) NOT NULL,\n"
                 + "  payload %2$s NOT NULL,\n"
                 + "  PRIMARY KEY (id),\n"
-                + "  UNIQUE (sequence_name, sequence_prev_id)\n"
+                + "  UNIQUE (sequence_name_prev_id_sha1)\n"
```

Comment:
No there's no guard. It's ensured by http://git-scm.com/book/en/v1/Git-Tools-Revision-Selection#A-SHORT-NOTE-ABOUT-SHA-1 :)